flowmachine/tests/test_cache_utils.py

```diff
@@ -410,3 +413,59 @@ def test_cache_table_exists(flowmachine_connect):
     assert cache_table_exists(
         flowmachine_connect, daily_location("2016-01-01").store().result().md5
     )
+
+
+def test_redis_resync(flowmachine_connect):
+    """
+    Test that redis states can be resynced to the flowdb cache.
+    """
+    stored_query = daily_location("2016-01-01").store().result()
+    assert (
+        QueryStateMachine(Table.redis, stored_query.md5).current_query_state
+        == QueryState.COMPLETED
+    )
+    assert stored_query.is_stored
+    Table.redis.flushdb()
+    assert stored_query.is_stored
+    assert (
+        QueryStateMachine(Table.redis, stored_query.md5).current_query_state
+        == QueryState.KNOWN
+    )
+    resync_redis_with_cache(flowmachine_connect, Table.redis)
+    assert (
+        QueryStateMachine(Table.redis, stored_query.md5).current_query_state
+        == QueryState.COMPLETED
+    )
+
+
+def test_cache_reset(flowmachine_connect):
+    """
+    Test that cache and redis are reset.
+    """
+    stored_query = daily_location("2016-01-01").store().result()
+    assert (
+        QueryStateMachine(Table.redis, stored_query.md5).current_query_state
+        == QueryState.COMPLETED
+    )
+    assert stored_query.is_stored
+    reset_cache(flowmachine_connect, Table.redis)
+    assert (
+        QueryStateMachine(Table.redis, stored_query.md5).current_query_state
+        == QueryState.KNOWN
+    )
+    assert not stored_query.is_stored
+
+
+def test_redis_resync_runtimeerror(flowmachine_connect, dummy_redis, monkeypatch):
+    """
+    Test that a runtime error is raised if redis is being updated from multiple places when trying to resync.
+    """
+    monkeypatch.setattr("flowmachine.core.query.Query.redis", dummy_redis)
```

Comment:
I may have asked this before, but is it possible/sensible to move the monkeypatching into the `dummy_redis` fixture itself?